### PR TITLE
Add support for apps webhook config endpoints

### PR DESCRIPTION
**Pull request by rpelliard**
This adds support for two Apps API endpoints (see #2095)
- [Get a webhook configuration for an app](https://docs.github.com/en/rest/reference/apps#get-a-webhook-configuration-for-an-app)
- [Update a webhook configuration for an app](https://docs.github.com/en/rest/reference/apps#update-a-webhook-configuration-for-an-app)

Both endpoints require authentication as an app (with a JWT), which can be dealt with by the underlying transport.

A few notes:
- I reused / extended the `HookConfig` struct which was defined in `orgs_audit_logs.go`, since it represents the same object.

- The `InsecureSSL` field of the struct is a `*string`. However, GitHub only returns `"0"` or `"1"` (insecure SSL disabled / enabled), so I think it'd be more user-friendly to use a `*bool`, and write some logic so that boolean values are correctly (un)marshaled from `"0"`/`"1"`. I'd be happy to write the code to deal with this, but that would be a breaking change so I wanted to check with you first.

- This is probably out of scope of this PR, but the [`Hook` struct](https://github.com/google/go-github/blob/master/github/repos_hooks.go#L87) used for repository / organization webhooks endpoints is using a `map[string]interface{}` for the `Config` field, but I think it could be using the `HookConfig` struct as well since it represents the same object. Happy to make that change in a separate PR, but this would be a breaking change as well.

### CODE REVIEW

File: github/orgs_audit_log.go
Comment (gmlewis) [inline github/orgs_audit_log.go:28]:
```suggestion
	// Secret is returned obfuscated by GitHub, but it can be set for outgoing requests.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @rpelliard !
One tiny nit to fix, please, but otherwise LGTM.

Awaiting second LGTM before merging.

**Review by gmlewis — APPROVED** [review]
LGTM.

Awaiting second LGTM before merging.

**Comment by rpelliard** [comment]
Hello @gmlewis @wesleimp, do you know when we'd be able to merge this ?

**Comment by gmlewis** [comment]
> Hello @gmlewis @wesleimp, do you know when we'd be able to merge this ?

Maybe @Parker77 has time to take a look at this one.

File: github/apps_hooks.go
Comment (gmlewis) [inline github/apps_hooks.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

No need for action... I'll add this required header.

File: github/apps_hooks_test.go
Comment (gmlewis) [inline github/apps_hooks_test.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

No need for action... I'll add this required header.

File: github/doc.go
Comment (Parker77) [inline github/doc.go:76]:
```suggestion
It supports both authentication as an installation, using an installation access token,
```

**Review by Parker77 — APPROVED** [review]
One minor nit, otherwise LGTM.

**Comment by Parker77** [comment]
@googlebot I fixed it.

**Comment by gmlewis** [comment]
> One minor nit, otherwise LGTM.

Thank you, @Parker77 !
Merging.

**Pull request merged**
